plugin/mqtt/client.go

```diff
@@ -219,9 +234,9 @@ func (m *Client) listen(topic string) paho.Token {
 }
 
 // WaitForToken synchronously waits until token operation completed
-func (m *Client) WaitForToken(action, topic string, token paho.Token) {
-	if inflight := atomic.LoadUint32(&m.inflight); inflight > 64 {
-		return
+func (m *Client) waitForToken(action, topic string, token paho.Token) {
+	if inflight := atomic.LoadUint32(&m.inflight); inflight > parallelInflightLimit {
```

Comment:
why do we still need this is the Sema does the rate limiting?

Reply:
removed the atomic, did nothing helpful and is not longer needed.
also moved the waitForToken code directly to the `Publish` since its the only place where its used.